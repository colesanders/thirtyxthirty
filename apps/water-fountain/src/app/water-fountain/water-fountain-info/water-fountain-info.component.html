<mat-card class="mat-card-dark"> 
    <mat-card-header style="display: flex; justify-content: space-between;">
      <mat-card-title class="accent">
        <span *ngIf="waterFountainForm.value.id !== null; else elseBlock">
          Editing WaterFountain
        </span>
        <ng-template #elseBlock>
          Create WaterFountain
        </ng-template>
        <mat-icon svgIcon="thirty-water-drop"></mat-icon>
      </mat-card-title>
      
    </mat-card-header>
  <form [formGroup]="waterFountainForm" (ngSubmit)="saved.emit()">
    <mat-card-content>

      <mat-form-field class="full-width" style="width: 50%" >
        <mat-label *ngIf="waterFountainForm.invalid && waterFountainForm.touched">
          Name must contain at least 2 characters.
        </mat-label>
        <input  matInput required placeholder="Name" type="string"
        formControlName="name">
      </mat-form-field>
      <br>

      <mat-form-field class="full-width">
        <mat-label *ngIf="waterFountainForm.invalid && waterFountainForm.touched">
          Description must contain 10 characters.
        </mat-label>

        <textarea  matInput required placeholder="Description" type="text"
        formControlName="description">
        </textarea>
      </mat-form-field>
      <br>


    </mat-card-content>
    <mat-card-actions style="display: flex; justify-content: space-between;">
      <button [disabled]="waterFountainForm.invalid" type="submit" mat-button color="primary">Save</button>
      <button type="button" mat-button (click)="cancelled.emit();waterFountainForm.reset();">
        Cancel
      </button>
    </mat-card-actions>
  </form>
  {{ waterFountainForm.value | json}}
</mat-card>