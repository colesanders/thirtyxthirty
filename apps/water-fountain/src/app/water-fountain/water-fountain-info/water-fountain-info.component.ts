import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { WaterFountain } from '@thirty/api-interfaces';
import {FormGroup } from '@angular/forms';

//import React from 'react';
import Rating from '@material-ui/lab/Rating';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';

@Component({
  selector: 'thirty-water-fountain-info',
  templateUrl: './water-fountain-info.component.html',
  styleUrls: ['./water-fountain-info.component.scss']
})
export class WaterFountainInfoComponent implements OnInit {
  @Input() waterFountainForm;
  @Output() saved = new EventEmitter();
  @Output() cancelled = new EventEmitter();

  starDisplay: number[];
  

  constructor() { }

  ngOnInit(): void {
    this.starDisplay = [1,1,1,1,1];
  }
}
