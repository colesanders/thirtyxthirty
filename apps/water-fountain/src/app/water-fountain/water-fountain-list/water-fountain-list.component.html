<mat-card class="mat-card-dark">
    <mat-card-header>
      <mat-card-title class="accent">
        WaterFountains List
      </mat-card-title>
    </mat-card-header>
    <mat-card-content>
      <mat-list>
        <mat-list-item *ngFor="let waterFountain of waterFountains" (click)="selected.emit(waterFountain)" style="cursor: pointer">
          <h3 matLine> {{ waterFountain.name | titlecase }} </h3>
          <p matLine>
            {{ waterFountain.description }}
          </p>
          <button mat-icon-button color="warn" (click)="deleted.emit(waterFountain); $event.stopImmediatePropagation()" >
            <mat-icon>delete</mat-icon>
          </button>
        </mat-list-item>
      </mat-list>
      
    </mat-card-content>
</mat-card>