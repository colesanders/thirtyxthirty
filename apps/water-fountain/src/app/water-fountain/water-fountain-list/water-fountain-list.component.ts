import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { WaterFountain } from '@thirty/api-interfaces';

@Component({
  selector: 'thirty-water-fountain-list',
  templateUrl: './water-fountain-list.component.html',
  styleUrls: ['./water-fountain-list.component.scss']
})
export class WaterFountainListComponent implements OnInit {
  @Input() waterFountains: [WaterFountain];
  @Output() selected = new EventEmitter<WaterFountain>();
  @Output() deleted = new EventEmitter<WaterFountain>();
  constructor() { }

  ngOnInit(): void {
  }

}
