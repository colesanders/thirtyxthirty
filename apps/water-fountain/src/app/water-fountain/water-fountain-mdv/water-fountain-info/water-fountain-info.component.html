<mat-card class="mat-card-dark"> 
    <mat-card-header style="display: flex; justify-content: space-between;">
      <mat-card-title class="accent">
        <span *ngIf="waterFountainForm.value.id !== null; else elseBlock">
          Editing WaterFountain
        </span>
        <ng-template #elseBlock>
          Create WaterFountain
        </ng-template>
        <mat-icon svgIcon="thirty-water-drop"></mat-icon>
      </mat-card-title>
      
    </mat-card-header>
  <form [formGroup]="waterFountainForm" (ngSubmit)="saved.emit()">
    <mat-card-content>

      <mat-form-field class="full-width" style="width: 50%" >
        <mat-label *ngIf="waterFountainForm.invalid && waterFountainForm.touched">
          Name must contain at least 2 characters.
        </mat-label>
        <input  matInput required placeholder="Name" type="string"
        formControlName="name">
      </mat-form-field>
      <br>

      <mat-form-field class="full-width">
        <mat-label *ngIf="waterFountainForm.invalid && waterFountainForm.touched">
          Description must contain 10 characters.
        </mat-label>

        <textarea  matInput required placeholder="Description" type="text"
        formControlName="description">
        </textarea>
      </mat-form-field>
      <br>

      <section>
        <mat-slider formControlName="pressure"
        thumbLabel tickInterval="1" min="1" max="5">
        </mat-slider>

        <mat-slider formControlName="temperature"
        thumbLabel tickInterval="1" min="1" max="5">
        </mat-slider>

        <mat-label>Average Water Rating</mat-label><br>
        <span *ngFor="let drop of dropDisplay | slice:0: (waterFountainForm.value.pressure + waterFountainForm.value.temperature)/2">
          <mat-icon svgIcon="thirty-water-drop" class="star-color"></mat-icon>
        </span>
      </section>


    </mat-card-content>
    <mat-card-actions style="display: flex; justify-content: space-between;">
      <button [disabled]="waterFountainForm.invalid" type="submit" mat-button color="primary">Save</button>
      <button type="button" mat-button (click)="cancelled.emit();waterFountainForm.reset();">
        Cancel
      </button>
    </mat-card-actions>
  </form>
</mat-card>