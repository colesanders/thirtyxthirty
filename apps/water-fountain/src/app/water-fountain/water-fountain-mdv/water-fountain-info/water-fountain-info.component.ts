import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { WaterFountain } from '@thirty/api-interfaces';


@Component({
  selector: 'thirty-water-fountain-info',
  templateUrl: './water-fountain-info.component.html',
  styleUrls: ['./water-fountain-info.component.scss']
})
export class WaterFountainInfoComponent implements OnInit {
  @Input() waterFountainForm;
  @Output() saved = new EventEmitter();
  @Output() cancelled = new EventEmitter();

  dropDisplay: number[];
  

  constructor() { }

  ngOnInit(): void {
    this.dropDisplay = [1,1,1,1,1];
  }
}
