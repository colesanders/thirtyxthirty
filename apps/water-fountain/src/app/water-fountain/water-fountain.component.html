<div class="waterFountain-containers">

    <div class="waterFountain-list">
        <thirty-water-fountain-list [waterFountains]="waterFountain$ | async" (selected)="selectWaterFountain($event)" (deleted)="deleteWaterFountain($event)">
        </thirty-water-fountain-list>
    </div>
    <div class="waterFountain-info">
        <thirty-water-fountain-info [waterFountainForm]="waterFountainForm" (saved)="saveWaterFountain()" (cancelled)="resetForm()"> 
        </thirty-water-fountain-info>
    </div>
</div>
