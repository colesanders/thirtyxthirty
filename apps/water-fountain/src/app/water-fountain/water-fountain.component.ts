import { Component, OnInit } from '@angular/core';
import { WaterFountain } from '@thirty/api-interfaces';
import { WaterFountainService, SnackBarService } from '@thirty/core-data';
import { Observable } from 'rxjs';
import { FormGroup, FormBuilder, FormControl, Validators } from '@angular/forms';

@Component({
  selector: 'thirty-water-fountain',
  templateUrl: './water-fountain.component.html',
  styleUrls: ['./water-fountain.component.scss']
})
export class WaterFountainComponent implements OnInit {
  title = 'WaterFountain Store';
  selectedWaterFountain: WaterFountain;
  waterFountainForm: FormGroup
  waterFountain$: Observable<WaterFountain[]>;

  constructor(private waterFountainService: WaterFountainService, private formBuilder: FormBuilder, public snackBarService: SnackBarService) { }

  ngOnInit(): void {
    this.creatFormGroup();
    this.refresh();
  }

  deleteWaterFountain(waterFountain:WaterFountain) {
    this.waterFountainService.deleteWaterFountain(waterFountain)
    .subscribe( m => {
      this.refresh();
    });
  }

  saveWaterFountain() {
    const waterFountain = this.waterFountainForm.value;

    this.snackBarService.openSnackBar("WaterFountain Saved!", "Okay", 2000)

    if(waterFountain.id !== null){
      this.waterFountainService.updateWaterFountain(waterFountain)
      .subscribe( m => {
        this.refresh();
      });
    } else {
      this.waterFountainService.createWaterFountain(waterFountain)
      .subscribe( m => {
        this.refresh();
      });
    }
  }

  refresh(){
    this.loadWaterFountains();
    this.resetForm();
  }

  creatFormGroup(){
    this.waterFountainForm = this.formBuilder.group({
      id: [],

      name: new FormControl('', [
        Validators.required,
        Validators.minLength(2),
      ]),
      description: new FormControl('', [
        Validators.required,
        Validators.minLength(10),
      ]),
      pressure: new FormControl(0, [
        Validators.required,
      ]),
      temperature: new FormControl(0, [
        Validators.required,
      ]),
      stars: new FormControl(1, [
        Validators.required,
      ]),
    })
  }

  resetForm() {
    this.waterFountainForm.patchValue({});
  }

  selectWaterFountain(waterFountain: WaterFountain) {
    this.waterFountainForm.patchValue(waterFountain);
  }

  loadWaterFountains(): void{
    this.waterFountain$ = this.waterFountainService.getWaterFountains();
  }

}
