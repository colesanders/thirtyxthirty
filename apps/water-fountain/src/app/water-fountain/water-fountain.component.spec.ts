import { HttpClientModule } from '@angular/common/http';
import { DebugElement } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { WaterFountainService } from '@thirty/core-data';
import { MaterialModule } from '@thirty/material';
import { of } from 'rxjs';
import { WaterFountainComponent } from './water-fountain.component';
import { WaterFountain } from '@thirty/api-interfaces';
import { WaterFountainInfoComponent } from './water-fountain-info/water-fountain-info.component';
import { WaterFountainListComponent } from './water-fountain-list/water-fountain-list.component';

const mockEmptyWaterFountain: WaterFountain = {
  id: null,
  name: 'Mock Empty WaterFountain',
  description: '',
  pressure: 0,
  temperature: 0,
  stars: 1,
};

const mockWaterFountain : WaterFountain = {
  id: 1,
  name: 'Mock WaterFountain',
  description: 'Mock',
  pressure: 1,
  temperature: 1,
  stars: 1,
};

const mockWaterFountainService = {
  getWaterFountains: () => of([]),
  createWaterFountain: () => of({}),
  updateWaterFountain: () => of({}),
  deleteWaterFountain: () => of({})
}

describe('WaterFountainComponent', () => {
  let component: WaterFountainComponent;
  let fixture: ComponentFixture<WaterFountainComponent>;
  let de: DebugElement;
  let waterFountainService: WaterFountainService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        HttpClientModule,
        MaterialModule,
        NoopAnimationsModule,
        RouterTestingModule,
      ],
      providers: [
        { provide: WaterFountainService, useValue: mockWaterFountainService }
      ],
      declarations: [
        WaterFountainComponent,
        WaterFountainInfoComponent,
        WaterFountainListComponent
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(WaterFountainComponent);
    component = fixture.componentInstance;
    de = fixture.debugElement;
    waterFountainService = de.injector.get(WaterFountainService);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should select a waterFountain', () => {
    component.selectWaterFountain(mockWaterFountain);
    expect(component.waterFountainForm.value).toMatchSnapshot();
  });
});
