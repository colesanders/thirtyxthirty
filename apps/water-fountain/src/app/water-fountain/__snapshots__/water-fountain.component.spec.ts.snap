// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaterFountainComponent should select a waterFountain 1`] = `
Object {
  "description": "Mock",
  "id": 1,
  "name": "Mock WaterFountain",
  "pressure": 1,
  "stars": 1,
  "temperature": 1,
}
`;
